src/App.tsx

```diff
@@ -1,26 +1,209 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  addTodos,
+  deleteTodos,
+  getTodos,
+  updateTodos,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Header } from './components/header';
+import { TodoList } from './components/todoList';
+import { Footer } from './components/footer';
+import { Sort } from './types/Sort';
+import { ErrorMassage } from './components/errorMassage';
+import { ErrorMessage } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const normalize = (todos: Todo[]): Todo[] =>
+  todos.map(t => ({ ...t, completed: t.completed === true }));
 
 export const App: React.FC = () => {
+  const [userTodos, setUserTodos] = useState<Todo[]>([]);
+  const [errorMassage, setErrorMassage] = useState<ErrorMessage | null>(null);
+  const [errorKey, setErrorKey] = useState(0);
+  const [sortTodo, setsortTodo] = useState<Sort>('all');
+  const [clearCompletedDisabled, setClearCompletedDisabled] = useState(true);
+  const [isLoader, setIsLoader] = useState<string | number | null>(null);
+  const [tempTitle, setTempTitle] = useState('');
+  const [updateAfterClearDelete, setUpdateAfterClearDelete] = useState(true);
+  const [focusSignal, setFocusSignal] = useState(0);
+
+  const hideErrorNow = useCallback(() => {
+    setErrorMassage(null);
+  }, []);
+
+  const showError = useCallback((msg: ErrorMessage) => {
+    setErrorMassage(msg);
+    setErrorKey(k => k + 1);
+  }, []);
+
+  useEffect(() => {
+    getTodos()
+      .then(ts => setUserTodos(normalize(ts)))
+      .catch(() => showError(ErrorMessage.Load));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [updateAfterClearDelete]);
+
+  useEffect(() => {
+    setClearCompletedDisabled(!userTodos.some(t => t.completed));
+  }, [userTodos]);
```

Comment:
```suggestion

  const isClearCompletedDisabled =  !userTodos.some(t => t.completed);

```

no need extra state (and extra re renders). just count this value based on userTodos.